File: level1/09. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기_박병주.js
function solution(strings, n) {
    var answer = [];
    var charArr = [];
    
    strings.sort();
    for(var i=0; i<strings.length; i++){

        charArr.push(strings[i].charAt(n));
    }
    
    charArr.sort();
    
    for(var i=0; i<charArr.length;i++){
        
        for(var j=0; j<strings.length;j++){
            
            if( strings[j].charAt(n) == charArr[i]){
                //console.log("strings : "+strings[j]+", char : "+charArr[i]);
                answer.push(strings[j]);
                strings.splice(j,1);
                break;
            }
        }
    }

    return answer;
}

File: level1/09. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기_박병주.java
import java.util.*;

class Solution {
    public List solution(String[] strings, int n) {
        List answer = new ArrayList<>();
        List nthChar = new ArrayList<>();
        
        for(int i=0; i<strings.length;i++){
            System.out.print(strings[i]+", ");
        }
               
        Arrays.sort(strings);
        
        for(int i=0; i<strings.length;i++){
            System.out.print(strings[i]+", ");
        }
                
        for(int i=0; i<strings.length; i++){
            System.out.println(strings[i]);
            nthChar.add(strings[i].charAt(n));
        }
        
        Collections.sort(nthChar);
        
        for(int i=0; i<nthChar.size();i++){
            for(int j=0; j<strings.length;j++){
                if(strings[j] == null ) continue;
                if( nthChar.get(i).equals(strings[j].charAt(n))){
                    answer.add(strings[j]);
                    strings[j] = null;
                }
            }            
        }
        return answer;
    }
}